internal/controller/managedcluster_controller.go

```diff
@@ -365,6 +373,10 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, l logr.Logger, ma
 			return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
 		}
 
+		if !managedCluster.Status.CCMSecretReconciled {
```

Comment:
if we update credentials, new secret won't be propagated. if you want to skip this part when it was reconciled successfully, you should track both `name` and `generation` of the credential object (+ underlying ClusterIdentity reference could be updated with no sign on the credential object right now)

Reply:
Yep, this is a problem. I will rewrite that part, and probably get rid of the `CCMSecretReconciled` field completely

Reply:
@Kshatrix @zerospiel I removed the `CCMSecretReconciled` variable completely and now secrets will be applied on the managed cluster on every reconcile.

I would like to actually not do that, but as Andrey mentioned earlier it will require some changes in Credential controller to properly track changes. 

So if you're ok with the each reconcile thing we can leave it as is for now and I will create task for tracking credential resources properly. If not I will have to do the tracking task first and I would prefer to do it in a separate PR (this one is big enough already)

Reply:
Sounds ok to me 👌

Reply:
Follow up task on that #529